### PR TITLE
Make sweeper for google_vmwareengine_cluster handwritten, add iteration over loop of regions/locations

### PR DESCRIPTION
**Pull request by SarahFrench**
PR is related to:
* google_vmwareengine_private_cloud  sweeper changes https://github.com/GoogleCloudPlatform/magic-modules/pull/11002
* google_vmwareengine_network  sweeper changes https://github.com/GoogleCloudPlatform/magic-modules/pull/10994

This sweeper may not be 100% necessary, as the sweeper for Private Clouds force deletes them, i.e. removes all resources inside including clusters: https://github.com/GoogleCloudPlatform/magic-modules/pull/11002/files#r1647648806

The generated sweeper isn't functional though, so this PR could pivot to removing the generated sweeper only.

```release-note:none

```

### CODE REVIEW

File: mmv1/products/vmwareengine/Cluster.yaml [hunk 1]
```diff
@@ -49,6 +49,7 @@ async: !ruby/object:Api::OpAsync
 import_format: ["{{%parent}}/clusters/{{name}}"]
 id_format: "{{parent}}/clusters/{{name}}"
 autogen_async: true
+skip_sweeper: true
```
Comment (SarahFrench) [inline mmv1/products/vmwareengine/Cluster.yaml:52]:
```suggestion
# There is a handwritten sweeper that provides a list of locations to sweep
skip_sweeper: true
```

File: mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_cluster_sweeper.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	}
+
+	t := &testing.T{}
+	billingId := envvar.GetTestBillingAccountFromEnv(t)
+
+	// List of location values includes:
+	//   * global location
+	//   * regions used for this resource type's acc tests in the past
+	//   * the 'region' passed to the sweeper
+	locations := []string{"global", "us-central1", "southamerica-west1", "me-west1", region}
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_cluster_sweeper.go:43]:
```suggestion
	locations := []string{region, "global", "southamerica-west1", "me-west1"}
```

Use user-provided region first, remove us-central1 as it's assumed that's the value of region most times

File: mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_network_sweeper.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+	}
+
+	t := &testing.T{}
+	billingId := envvar.GetTestBillingAccountFromEnv(t)
+
+	// List of location values includes:
+	//   * global location
+	//   * regions used for this resource type's acc tests in the past
+	//   * the 'region' passed to the sweeper
+	locations := []string{"global", "us-central1", "southamerica-west1", "me-west1", region}
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_network_sweeper.go:43]:
```suggestion
	locations := []string{region, "global", "southamerica-west1", "me-west1"}
```

Use user-provided region first, remove us-central1 as it's assumed that's the value of region most times

File: mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_private_cloud_sweeper.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+	}
+
+	t := &testing.T{}
+	billingId := envvar.GetTestBillingAccountFromEnv(t)
+
+	// List of location values includes:
+	//   * global location
+	//   * regions used for this resource type's acc tests in the past
+	//   * the 'region' passed to the sweeper
+	locations := []string{"global", "us-central1", "southamerica-west1", "me-west1", region}
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_private_cloud_sweeper.go:44]:
```suggestion
	locations := []string{region, "global", "southamerica-west1", "me-west1"}
```

Use user-provided region first, remove us-central1 as it's assumed that's the value of region most times

File: mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_cluster_sweeper.go [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+		return err
+	}
+
+	t := &testing.T{}
+	billingId := envvar.GetTestBillingAccountFromEnv(t)
+
+	// List of location values includes:
+	//   * zones used for this resource type's acc tests in the past
+	//   * the 'region' passed to the sweeper
+	locations := []string{region, "us-central1-a", "southamerica-west1-a", "me-west1-a"}
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_cluster_sweeper.go:43]:
```suggestion
	locations := []string{region, "us-central1-a", "us-central1-b", "southamerica-west1-a", "southamerica-west1-b", "me-west1-a", "me-west1-b"}
```

**Review by melinath — APPROVED** [review]

**Comment by melinath** [comment]
Have you already tested this manually?

**Comment by SarahFrench** [comment]
Yeah - There's [this build](https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleCloud_GOOGLE_MMUPSTREAMTESTS_SERVICE_SWEEPER__MANUAL/171682?buildTab=log&focusLine=0&logView=flowAware&linesState=1147) that includes logs like:

```
18:03:55   2024/06/20 17:03:55 [INFO][SWEEPER_LOG] Sweeping will include these locations: [us-central1 us-central1-a southamerica-west1-a me-west1-a].
18:03:55   2024/06/20 17:03:55 [INFO][SWEEPER_LOG] Beginning the process of sweeping location 'us-central1'.
18:03:55   2024/06/20 17:03:55 [INFO][SWEEPER_LOG] looking for parent resources in location 'us-central1'.
```

**Comment by melinath** [comment]
Great, thanks for confirming!

**Comment by SarahFrench** [comment]
Here's a better example run that manages to find the parent private cloud resources but isn't able to find any clusters in them: https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleCloud_GOOGLE_MMUPSTREAMTESTS_SERVICE_SWEEPER__MANUAL/176420?buildTab=log&focusLine=1541&logView=linear&linesState=1131

I haven't technically seen the sweeper successfully remove a cluster yet, as haven't run it when the VCR project is in the correct state for that, but the sweeper prior to this PR was broken so merging as-is won't break anything!

**Pull request merged**